src/modules/chat/chat.controller.ts

```diff
@@ -54,14 +53,24 @@ export class ChatController {
     return await this.chatService.getChat(chatId);
   }
 
-  @Put('/update')
-  async updateChat(@Query('uid') uid, @Body() createChatDTO: CreateChatDTO) {
-    return await this.chatService.updateChat(uid, createChatDTO);
+  @Put('/:chatId')
+  async updateChat(
+    @Param('chatId') chatId: string,
+    @Body() createChatDTO: CreateChatDTO,
```

Comment:
Why `createChatDto`, create a new DTO for chat updation.

Reply:
> Create an chat architecture on paper or software of your choice, see how the chats should be organised, the datatypes of each of the fields, and how to distinguish between our written message and others.
> 
> Currently, there are problems with the chosen datatypes and some of the logic that are applied here.

Thanks
Another question is that.. will we have a thread based chat structure? like in stack overflow?
or a normal linear chat structure where every user can create a chat in the course discussion?